plugins/modules/s3_object.py

```diff
@@ -917,6 +954,243 @@ def ensure_tags(client, module, bucket, obj):
     return current_tags_dict, changed
 
 
+def s3_object_do_get(module, connection, s3_vars):
+
+    keyrtn = key_check(module, connection, s3_vars.bucket, s3_vars.object, version=s3_vars.version, validate=s3_vars.validate)
+    if keyrtn is False:
+        if s3_vars.version:
+            module.fail_json(msg="Key %s with version id %s does not exist." % (s3_vars.object, s3_vars.version))
+        else:
+            module.fail_json(msg="Key %s does not exist." % s3_vars.object)
+    if s3_vars.dest and path_check(s3_vars.dest) and s3_vars.overwrite != 'always':
+        if s3_vars.overwrite == 'never':
+            module.exit_json(msg="Local object already exists and overwrite is disabled.", changed=False)
+        if s3_vars.overwrite == 'different' and etag_compare(module, connection, s3_vars.bucket, s3_vars.object, version=s3_vars.version, local_file=s3_vars.dest):
+            module.exit_json(msg="Local and remote object are identical, ignoring. Use overwrite=always parameter to force.", changed=False)
+        if s3_vars.overwrite == 'latest' and is_local_object_latest(module, connection, s3_vars.bucket, s3_vars.object, version=s3_vars.version, local_file=s3_vars.dest):
+            module.exit_json(msg="Local object is latest, ignoreing. Use overwrite=always parameter to force.", changed=False)
+
+    try:
+        download_s3file(module, connection, s3_vars.bucket, s3_vars.object, s3_vars.dest, s3_vars.retries, version=s3_vars.version)
+    except Sigv4Required:
+        connection = get_s3_connection(module, s3_vars.aws_connect_kwargs, s3_vars.location, s3_vars.ceph, s3_vars.endpoint_url, sig_4=True)
+        download_s3file(module, connection, s3_vars.bucket, s3_vars.obj, s3_vars.dest, s3_vars.retries, version=s3_vars.version)
+
+    module.exit_json(failed=False)
+
+
+def s3_object_do_put(module, connection, s3_vars):
+    # if putting an object in a bucket yet to be created, acls for the bucket and/or the object may be specified
+    # these were separated into the variables bucket_acl and object_acl above
+
+    if s3_vars.content is None and s3_vars.content_base64 is None and s3_vars.src is None:
+        module.fail_json(msg='Either content, content_base64 or src must be specified for PUT operations')
+    if s3_vars.src is not None and not path_check(s3_vars.src):
+        module.fail_json(msg='Local object "%s" does not exist for PUT operation' % (s3_vars.src))
+
+    keyrtn = None
+    if s3_vars.bucketrtn:
+        keyrtn = key_check(module, connection, s3_vars.bucket, s3_vars.object, version=s3_vars.version, validate=s3_vars.validate)
+    else:
+        # If the bucket doesn't exist we should create it.
+        # only use valid bucket acls for create_bucket function
+        s3_vars.permission = s3_vars.bucket_acl
+        create_bucket(module, connection, s3_vars.bucket, s3_vars.location)
+
+    # the content will be uploaded as a byte string, so we must encode it first
+    bincontent = None
```

Comment:
Suggestion: You can move the 5 lines related to `bincontent` in a dedicated function.